### PR TITLE
Introduced the following new elements.

### PR DESCRIPTION
**Pull request by DHirani**
1. logname - you can override the name of the log file rather than using the EXE name, this means you don't have to call your EXE a different name, just name the winsw exe different. Default's the name to the EXE as before.
2. outfiledisabled - you can disable writing to the out file. Defaults to false.
3. errfiledisabled - you can disable writing to the error file. Defaults to false.
4. outfilepattern - you can choose the pattern of the out file. Defaults to .out.log.
5. errfilepattern - you can choos the pattern of the error file. Defaults to .err.log.

### CODE REVIEW

**Comment by BuddyWork** [comment]
1 logname - you can override the name of the log file rather than using the EXE name, this means you don't have to call your EXE a different name, just name the winsw exe different. Default's the name to the EXE as before.
2. outfiledisabled - you can disable writing to the out file. Defaults to false.
3. errfiledisabled - you can disable writing to the error file. Defaults to false.
4. outfilepattern - you can choose the pattern of the out file. Defaults to .out.log.
5. errfilepattern - you can choos the pattern of the error file. Defaults to .err.log.

File: src/Core/WinSWCore/WinSWCore.csproj [hunk 1]
```diff
@@ -33,7 +33,7 @@
   </PropertyGroup>
   <PropertyGroup>
     <SignAssembly>true</SignAssembly>
-    <AssemblyOriginatorKeyFile>$(SolutionDir)..\winsw_key.snk</AssemblyOriginatorKeyFile>
+    <AssemblyOriginatorKeyFile>winsw_key.snk</AssemblyOriginatorKeyFile>
```
Comment (oleg-nenashev) [inline src/Core/WinSWCore/WinSWCore.csproj]: It does not work correctly on the solution level, at least in my IDE

File: src/Core/WinSWCore/Configuration/DefaultSettings.cs [hunk 2]
```diff
@@ -59,6 +59,11 @@ public string ExecutablePath
         public string LogDirectory { get { return Path.GetDirectoryName(ExecutablePath); } }
         public string LogMode { get { return "append"; } }
 
+        public bool OutFileDisabled { get { return false; } }
+        public bool ErrFileDisabled { get { return false; } }
+        public string OutFilePattern { get { return ".out.log"; } }
+        public string ErrFilePattern { get { return ".out.log"; } }
```
Comment (oleg-nenashev) [inline src/Core/WinSWCore/Configuration/DefaultSettings.cs]: Copy-paste error ?

File: src/Core/ServiceWrapper_dotNET4/manifest.xml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
+  <assemblyIdentity version="1.0.0.0" name="winsw" type="win32"/>
```
Comment (oleg-nenashev) [inline src/Core/ServiceWrapper_dotNET4/manifest.xml]: Version needs to be patched in the build flow if you want to include it to the release

File: src/Core/WinSWCore/ServiceDescriptor.cs [hunk 4]
```diff
@@ -355,6 +362,49 @@ public string LogMode
             }
         }
 
+        public string LogName
```
Comment (oleg-nenashev) [inline src/Core/WinSWCore/ServiceDescriptor.cs]: Imho all the options should be defined in the `<logging>` tag

File: src/Core/ServiceWrapper_dotNET4/winsw.xml [hunk 5]
```diff
@@ -0,0 +1,308 @@
+<!--
```
Comment (oleg-nenashev) [inline src/Core/ServiceWrapper_dotNET4/winsw.xml]: Why would you need this file?

File: src/Core/WinSWCore/WinSWCore.csproj [hunk 6]
```diff
@@ -73,6 +73,7 @@
   </ItemGroup>
   <ItemGroup>
     <None Include="packages.config" />
+    <None Include="winsw_key.snk" />
```
Comment (oleg-nenashev) [inline src/Core/WinSWCore/WinSWCore.csproj]: The SNK must not be included into the repo

File: src/Core/WinSWCore/WinSWCore.csproj (hunk 1 above)
Comment (BuddyWork) [reply src/Core/WinSWCore/WinSWCore.csproj]: Done

File: src/Core/WinSWCore/WinSWCore.csproj (hunk 6 above)
Comment (BuddyWork) [reply src/Core/WinSWCore/WinSWCore.csproj]: Done

File: src/Core/WinSWCore/Configuration/DefaultSettings.cs (hunk 2 above)
Comment (BuddyWork) [reply src/Core/WinSWCore/Configuration/DefaultSettings.cs]: Done

File: src/Core/ServiceWrapper_dotNET4/manifest.xml (hunk 3 above)
Comment (BuddyWork) [reply src/Core/ServiceWrapper_dotNET4/manifest.xml]: Removed the file should not been created.

File: src/Core/WinSWCore/ServiceDescriptor.cs (hunk 4 above)
Comment (BuddyWork) [reply src/Core/WinSWCore/ServiceDescriptor.cs]: I've followed the same logic as logmode, logpath, let me know if its wrong.

File: src/Core/ServiceWrapper_dotNET4/winsw.xml (hunk 5 above)
Comment (BuddyWork) [reply src/Core/ServiceWrapper_dotNET4/winsw.xml]: Removed.

**Comment by BuddyWork** [comment]
Removed the snk file and sqllite file

File: src/Core/WinSWCore/LogAppenders.cs [hunk 7]
```diff
@@ -299,16 +309,189 @@ private void CopyStreamWithRotation(Stream data, string ext)
...
+                    }
+                    catch (Exception e)
+                    {
+                        throw new IOException(string.Format("Failed to process file {0} due to error {1}",f, e.Message), e);
+                    }
+                }
+                if (nextFileNumber == 0) throw new IOException("Cannot roll the file because matching pattern not found");
+                nextFileNumber++;
+            }
+            return nextFileNumber;
```
Comment (oleg-nenashev) [inline src/Core/WinSWCore/LogAppenders.cs]: There is some copy-paste in this appender, but OK for now

File: src/Core/WinSWCore/ServiceDescriptor.cs (hunk 4 above)
Comment (oleg-nenashev) [reply src/Core/WinSWCore/ServiceDescriptor.cs]: ack, will need to think about it twice.

File: src/Core/WinSWCore/LogAppenders.cs (hunk 7 above)
Comment (DHirani) [reply src/Core/WinSWCore/LogAppenders.cs]: That is fine I will use my fork version until you merge it.

Comment (DHirani) [reply src/Core/WinSWCore/LogAppenders.cs]: Hi let me know when you planning to merge this into master.

Comment (oleg-nenashev) [reply src/Core/WinSWCore/LogAppenders.cs]: My optimistic plan - this Saturday

**Comment by oleg-nenashev** [comment]
Tested it a bit yesterday, looks good to me.
I will write some documentation for that before the release

**Comment by DHirani** [comment]
Thanks for pushing the change into master, when will the package be available from nuget?

**Comment by oleg-nenashev** [comment]
@DHirani it will be officially released once I update documentation to reflect the new features (README, logging page, config samples). It may take a while, but I will be happy to review the PRs if you have some time to work on them.

Anyway, the code looks good, so the documentation should not take much time

**Pull request merged**